core/txpool/legacypool/legacypool.go

```diff
@@ -823,6 +823,174 @@ func (pool *LegacyPool) add(tx *types.Transaction) (replaced bool, err error) {
 	return replaced, nil
 }
 
+// addAsync is a semi-async version of `add`. It frees up the global pool lock
+// to allow other functions to use the pending pool. It also performs some
+// operations on `pricedList` async to avoid waiting for reheap.
+//
+// It also validates a transaction and inserts it into the non-executable queue
+// for later pending promotion and execution. If the transaction is a replacement
+// for an already pending or queued one, it overwrites the previous transaction
+// if its price is higher.
+func (pool *LegacyPool) addAsync(tx *types.Transaction) (replaced bool, err error) {
```

Comment:
While this is a new function, most of the contents (except the lock changes) are same as `add`. While the diff doesn't detect it, I'd recommend reviewing this side by side with `add` function.

Reply:
Done in https://github.com/maticnetwork/bor/pull/1617/commits/5b84ffe57f5d43d4e2aac4f53fe5d876a16088f5